draft-ietf-taps-transport-security.md

```diff
@@ -249,11 +251,63 @@ interface or address changes.
 
 - Server: the peer responsible for responding to a session initiation.
 
+# Security Features
+
+In this section, we enumerate Security Features exposed by protocols discussed in the
+remainder of this document. Protocol security properties that are unrelated to the API surface
+exposed by such protocols, such as client or server identity hiding, are not listed 
+here as features.
+
+- Forward-secure key establishment: Cryptographic key establishment with forward secure properties.
```

Comment:
Is this feature related to the API surface exposed by security protocols? I think this might require a revision in the introductory paragraph, because ISTM this is a baseline feature rather than one motivated by the API.

Reply:
I think so. ESP, for example, does not provide this feature, and that might somehow be reflected in the API.

Reply:
That's mainly due to the fact that ESP doesn't do any handshake—all handshake protocols do this, right? So, one way to resolve this is to categorize the features by handshake and record protocols, and then things that don't affect the API (like this) for one of the categories doesn't need to be called out?

Reply:
Is the problem the distinction between record protocol and handshake? There's clearly a distinction between handshake and data transport in tcpcrypt: there is no renegotiation for instance (and the only control "message" is a single bit to indicate rekeying) so the handshake is well-defined in a way it isn't even for TLS.

Anyway, not necessarily advocating for this change. I think I'm satisfied that the PSK use case is significant enough to be useful to TAPS.